addons/product/models/product_template.py

```diff
@@ -644,17 +646,41 @@ def _get_attribute_exclusions(self, parent_combination=None):
             - has_dynamic_attributes: whether there is a dynamic attribute
             - no_variant_product_template_attribute_value_ids: values that are
                 no_variant
+           - parent_product_name: the name of the parent product if any, used in the interface
+               to explain why some combination are not available.
+               (e.g: Not available with Customizable Desk (Legs: Steel))
+           - mapped_attribute_names: the name of every attribute values based on their id,
+               used to explain in the interface why that combination is not available
+               (e.g: Not available with Color: Black)
         """
         self.ensure_one()
         return {
-            'exclusions': self._get_own_attribute_exclusions(),
+            'exclusions': self._complete_inverse_exclusions(self._get_own_attribute_exclusions()),
             'parent_exclusions': self._get_parent_attribute_exclusions(parent_combination),
             'archived_combinations': self._get_archived_combinations(),
             'has_dynamic_attributes': self.has_dynamic_attributes(),
             'existing_combinations': self._get_existing_combinations(),
             'no_variant_product_template_attribute_value_ids': self._get_no_variant_product_template_attribute_values(),
+            'parent_product_name': parent_name,
+            'mapped_attribute_names': self._get_mapped_attribute_names(parent_combination),
         }
 
+    @api.model
+    def _complete_inverse_exclusions(self, exclusions):
+        """Will complete the dictionnary of exclusions with their respective inverse
+        e.g: Black excludes XL and L
+        -> XL excludes Black
+        -> L excludes Black"""
+        result = dict(exclusions)
+        for key, value in exclusions.items():
+            for exclusion in value:
+                if exclusion in result and key not in result[exclusion]:
```

Comment:
I don't see how exactly?

You could already have other exclusions for "Large" attribute value (id 1).
Let's say it excludes "Red" (2) and "Blue" (3).

So it would look like `{1: [2, 3]}`

Then "Black" (4) excludes "Large", so it looks like:

```
{
    1: [2, 3],
    4: [1]
}
```

This method will add "Large" excludes "Black", like this:

```
{
    1: [2, 3, 4],
    4: [1]
}
```

(And also 2: [1] and 3: [1], but you get the point)

This is a little hard to explain so ask me if it doesn't make sense.